src/features/Console.ts

```diff
@@ -210,6 +211,17 @@ export class ConsoleFeature implements IFeature {
                     return;
                 }
 
+                if (vscode.window.activeTerminal.name !== "PowerShell Integrated Console") {
+                    this.log.write("PSIC is not active terminal. Running in active terminal using 'runSelectedText'");
+                    await vscode.commands.executeCommand("workbench.action.terminal.runSelectedText");
+
+                    // we need to honor the focusConsoleOnExecute setting here too.
+                    vscode.window.activeTerminal.show(!Settings.load().integratedConsole.focusConsoleOnExecute);
```

Comment:
So by calling `show()` it doesn't put focus on that console?

Reply:
The opposite. The parameter name is `preserveFocus` which, if `true` will keep the focus where it is. if `false` it will NOT preserver the focus and put focus on the terminal. That's why I needed to reverse the boolean.

Reply:
Is there a way to show the name of that parameter here? As it is, it's a bit hard to follow

Reply:
sadly there are [no named params in typescript](https://github.com/Microsoft/TypeScript/issues/467)... but I added a longer comment to explain why we need to inverse it.